src/plugins/vis_type_table/public/utils/use_pagination.ts

```diff
@@ -20,20 +20,23 @@ export const usePagination = (visConfig: TableVisConfig, nRow: number) => {
   ]);
 
   useEffect(() => {
-    const perPage = visConfig.perPage || 10;
+    const perPage = visParams.perPage || 0;
```

Comment:
I think I never explain this thoroughly but keeping changing this value. My bad. The line `const perPage = visParams.perPage || 0`; is setting the default value for `perPage` to 0 if `visParams.perPage` is not set or is a falsy value (e.g., null, undefined, or **0**). The reason for using 0 as the default value is because the pagination functionality is designed to be optional. When the value of perPage is set to 0, it indicates that pagination is not enabled, and the table should display all rows without pagination. If we set the default value to 10, it would mean that pagination would be enabled by default, displaying only 10 rows per page even if the user has not explicitly set a perPage value.

Reply:
So setting `const perPage = visParams.perPage || 0` makes logic sense but it makes the next line `const maxiPageIndex = Math.ceil(nRow / perPage) - 1;` vulnerable. There are two edge cases we should consider 1) `visParams.perPage` is 0. Well this is not possible. The `Max rows per page` option (src/plugins/vis_type_table/public/components/table_vis_options.tsx) forbids being set to 0. The `NumberInputOption` component for the `Max rows per page` option has a min property set to 1, which means the minimum allowed value for this input field is 1. 2) `visParams.perPage = ''`. This is possible if `Max rows per page` option is empty.  This would set maxiPageIndex to Infinity. The setPagination function will be called, but since maxiPageIndex is Infinity, the condition `p.pageIndex > maxiPageIndex` will always be false, and the pageIndex will not be updated. This is also reflected in the unit test:

```
it('should not set pagination if perPage is empty string', () => {
    visParams.perPage = '';
    const { result } = renderHook(() => usePagination(visParams, 20));
    expect(result.current).toEqual(undefined);
  });
```

However, the code still does not break. Even though maxiPageIndex becomes Infinity, the setPagination function will update the pageIndex value to be the same as the current pageIndex value, which means that the pagination state remains unchanged.

Reply:
Not necessary for approval, but I think this is a case where it just makes sense to abstract it out as a function, that can have its own test cases and comment explanations:

```
const getPerPageCount = (perPageConfig, nRow) => {
  // if no perPageConfig specified, set to 0 to disable pagination  
  const perPage =  perPageConfig || 0;
  return Math.min(perPage, nRow);
}
```